daemon/api.go

```diff
@@ -327,6 +332,22 @@ func getSnapsInfo(c *Command, r *http.Request) Response {
 		}
 	}
 
+	if len(includeTypes) > 0 {
+		for i, r := range results {
+			hasType := false
```

Comment:
This is super nitpicky so just ignore me if you feel its over the top. I would slightly prefer to split the "hasType" into a function. I find the loop easier to to read this way. Something like:
```
resultHasType := func(r map[string]interface{}, allowedTypes []string) bool {
		for _, t := range allowedTypes {
			if r["type"] == t {
				return true
			}
		}
		return false
	}
	if len(includeTypes) > 0 {
		for name, r := range results {
			if !resultHasType(r, includeTypes) {
				delete(results, name)
			}
		}
	}
```
(or instead of defining resultHasType in the function moving it out and make it a proper function).